pkg/util/azure/azure.go

```diff
@@ -0,0 +1,37 @@
+// Package azure adds additional capabilities similar to those found in
```

Comment:
I would not recommend using package named `azure`

Also why not using : https://github.com/Azure/go-autorest/blob/7439e774c9e95fcf068948c5ae3c78625cce6ffe/autorest/azure/azure.go#L176 ?
We already using this in `./hack/kubeadminkubeconfig/get.go` L20 for example ?

Reply:
What package name would you recommend? I named it that because `ParseResourceGroupID` is modeled after `ParseResourceID` in `go-autorest`, but if that's confusing I can rename it.

As for why I wrote a new function: `ParseResourceID` is for resource ID, not resource group ID. `/subscriptions/<ID>/resourceGroups/aro-example` isn't matched by `(?i)subscriptions/(.+)/resourceGroups/(.+)/providers/(.+?)/(.+?)/(.+)`.

Reply:
oh yea i see.

Reply:
As @jim-minter pointed out (https://github.com/Azure/ARO-RP/pull/151#discussion_r376858688), none of this code is even needed, so I've removed it.